addons/im_livechat/controllers/main.py

```diff
@@ -110,6 +112,7 @@ def get_session(self, channel_id, previous_operator_id=None, chatbot_script_id=N
 
         chatbot_script = operator_info['chatbot_script']
         is_chatbot_script = operator_info['operator_model'] == 'chatbot.script'
+        whitelisted_extra_channel_info = self.env['discuss.channel']._white_list_extra_channel_info(extra_channel_info)
```

Comment:
better use the term allow/deny rather than white/black

Also I would try to find another name than "extra_channel_info".
Maybe just `allowed_channel_fields`?

But this is not really an allow list, we should be doing processing for each case, such as validating access to agent, etc.
So the function should probably be named `_prepare_channel_fields` instead?
And the param `extra_channel_params`?